### PR TITLE
Update `step-security/harden-runner` configuration

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Update `step-security/harden-runner` configuration (#1177)

This resolves recent build failures by ensuring that JDKs can be
downloaded.

```

### CODE REVIEW

**Comment by Stephan202** [comment]
I force-pushed a change, but it's not showing up here yet :eyes:.

**Comment by Stephan202** [comment]
Still connection issues; I wonder whether this is a GitHub issue. Will check back later.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -31,8 +31,10 @@ jobs:
           disable-sudo: true
           egress-policy: block
           allowed-endpoints: >
+            api.adoptium.net:443
             github.com:443
             jitpack.io:443
+            objects.githubusercontent.com:433
```
Comment (varunsh-coder) [inline .github/workflows/build.yml:37]: I noticed the port number should be `443`, but is currently being set to `433` which is causing the outbound call to get blocked.

File: .github/workflows/pitest-analyze-pr.yml [hunk 2]
```diff
@@ -17,7 +17,9 @@ jobs:
           disable-sudo: true
           egress-policy: block
           allowed-endpoints: >
+            api.adoptium.net:443
             github.com:443
+            objects.githubusercontent.com:433
```
Comment (varunsh-coder) [inline .github/workflows/pitest-analyze-pr.yml:22]: Same here, the port should be `443`

File: .github/workflows/pitest-update-pr.yml [hunk 3]
```diff
@@ -25,8 +25,10 @@ jobs:
           disable-sudo: true
           egress-policy: block
           allowed-endpoints: >
+            api.adoptium.net:443
             api.github.com:443
             github.com:443
+            objects.githubusercontent.com:433
```
Comment (varunsh-coder) [inline .github/workflows/pitest-update-pr.yml:31]: Same here, the port should be `443`

File: .github/workflows/run-integration-tests.yml [hunk 4]
```diff
@@ -24,8 +24,10 @@ jobs:
           disable-sudo: true
           egress-policy: block
           allowed-endpoints: >
+            api.adoptium.net:443
             checkstyle.org:443
             github.com:443
+            objects.githubusercontent.com:433
```
Comment (varunsh-coder) [inline .github/workflows/run-integration-tests.yml:30]: Same here, the port should be `443`

**Comment by Stephan202** [comment]
Tnx @varunsh-coder! Thanks for pointing out my silly mistake. A+ support. :muscle: :heart:

**Comment by varunsh-coder** [comment]
> Tnx @varunsh-coder! Thanks for pointing out my silly mistake. A+ support. 💪 ❤️

No problem, feel free to tag me on GitHub or reach out to support@stepsecurity.io in case you have any questions or feedback.

**Review by mohamedsamehsalah — APPROVED** [review]
🔓

**Review by rickie — APPROVED** [review]

**Pull request merged**